source/clear-linux/guides/maintenance/assign-static-ip.rst

```diff
@@ -1,71 +1,175 @@
+************************************************
+
+New installations of |CL| use NetworkManager as the default network interface
+manager for all network connections.
+
+.. note:: 
+
+   * The *cloud* |CL| images continue to use systemd-networkd to manage
+     network connections.
+
+   * In earlier |CL| versions, systemd-network was used to manage Ethernet
+     interfaces and NetworkManager was used for wireless interfaces.
+
+
+Before defining a configuration for assigning a static IP address, you should
+verify which program is managing the network interface.
+
+#. Check the output of :command:`nmcli device` to see if NetworkManager is
+   managing the device.
+
+   .. code-block:: bash
+
+      nmcli device status
+
+   If the STATE column for the device shows *connected* or *disconnected*, the
+   network configuration is being managed by NetworkManager and the instructions
+   for :ref:`using NetworkManager <nm-static-ip>` should be used. 
+
+   If the STATE column for the device shows *unmanaged*, check to see if the
+   device is being managed by systemd-networkd  
+
+
+#. Check the output of :command:`networkctl list` to see if
+   systemd-networkd is managing the device.
+
+   .. code-block:: bash
+
+      networkctl list 
+
+   If the SETUP column for the device shows *configured*, the network
+   configuration is being managed by systemd-networkd and the instructions for
+   :ref:`using systemd-networkd <networkd-static-ip>` should be used. 
+
+
+.. _nm-static-ip:
+
+Using NetworkManager
+********************
+
+Network connections managed by NetworkManager are stored as files with the
+:file:`.nmconnection` file extension in the
+:file:`/etc/NetworkManager/system-connections/` directory.
+
+A few tools exists to aid to manipulate network connections managed by
+NetworkManager:
+
+* nmcli - a command-line tool 
+
+* nmtui - a text user interface that provides a pseudo graphical menu in the
+  terminal
+
+* nm-connection-editor - a graphical user interface
+
+The method below uses the command line tool *nmcli* to modify network
+connection. 
+
+
+#. Identify the existing connection name.
+
+   .. code:: bash
+
+      nmcli connection show
+
+   The output will look like this:
+
+   .. code:: bash
+
+      NAME                UUID                                  TYPE            DEVICE 
+      Wired connection 1  00000000-0000-0000-0000-000000000000  802-3-etherneten01
+
+   If a connection does not exist, it will need to be created with
+   :command:`nmcli connection add`.  
+
+
+#. Modify the connection to use a static IP address. Replace the variables in
+   brackets with the appropriate values. *[CONNECTION_NAME]* should be
+   replaced with the NAME from the command above. 
+
+   .. code::
+
+      sudo nmcli connection modify "[CONNECTION_NAME]" \
+      ipv4.method "manual" \
+      ipv4.addresses "[IP_ADDRESS]/[CIDR_NETMASK]" \
+      ipv4.gateway "[GATEWAY_IP_ADDRESS]" \
+      ipv4.dns "[PRIMARY_DNS_IP],[SECONDARY_DNS_IP]"
```

Comment:
@busykai thanks for the review! 

My understanding is that this line won't actually do anything because we're using systemd-resolved for DNS resolution, and we need to provide instructions to configure a [resolved.conf](https://www.freedesktop.org/software/systemd/man/resolved.conf.html). Can you confirm this is true?
```
      ipv4.dns "[PRIMARY_DNS_IP],[SECONDARY_DNS_IP]"
```

Reply:
@puneetse, `systemd-resolved` will get the DNS settings from `NetworkManager`. It would be proper to leave these instructions in place: setting static IP means lack of DHCP on the network in some cases and DNS will not be configured.

Alternatively, `systemd-resolved` can be configured to use particular DNS servers, as you mention. However, it's best not to do that and we should not instruct to do so.

Finally, `systemd-resolved` has "fallback" [well-known public] DNS servers (unless they are re-configured/disabled) which is why in absence any DNS configured in either NetworkManager or `systemd-resolved`, hostname resolution will still work.

In case of static IP configuration, it's preferable to configure DNS servers for the interface, because folks who configure static IP should also know the best (closest) DNS service to use for their network.